### PR TITLE
feature: PausableComposition feature flag in CCGP

### PR DESCRIPTION
**Pull request by zamulla**
Mentioning the PausableComposition flag in the Compose Compiler Gradle plugin reference: https://youtrack.jetbrains.com/issue/KT-71227/Compose-Add-PausableComposition-feature-flag-to-the-Compose-Gradle-Plugin

### CODE REVIEW

**Review by Tapchicoma — APPROVED** [review]

File: docs/topics/compose-compiler-options.md [hunk 1]
```diff
@@ -194,6 +194,27 @@ and functions that are implicitly not skippable (inline functions and functions
 >
 {style="warning"}
 
+### PausableComposition
+
+> The feature flag affects behavior only with a version of Compose runtime that supports pausable composition,
+> starting with `androidx.compose.runtime` 1.8.0-alpha02.
+> With older versions, the feature flag has no effect.
```
Comment (sarahhaggarty) [inline docs/topics/compose-compiler-options.md:201]:
```suggestion
> Older versions ignore the feature flag.
```

File: docs/topics/compose-compiler-options.md [hunk 2]
```diff
@@ -194,6 +194,27 @@ and functions that are implicitly not skippable (inline functions and functions
...
+>
+{style="note"}
+
+**Default**: disabled
+
+If enabled, changes code generation of composable functions to enable pausing when part of a pausable composition.
```
Comment (sarahhaggarty) [inline docs/topics/compose-compiler-options.md:207]:
```suggestion
If enabled, changes the code generation of composable functions to allow pausing when part of a pausable composition.
```

File: docs/topics/compose-compiler-options.md [hunk 3]
```diff
@@ -194,6 +194,27 @@ and functions that are implicitly not skippable (inline functions and functions
...
+This allows Compose runtime to suspend composition on the skipping points,
+letting long-running compositions be split across multiple frames.
```
Comment (sarahhaggarty) [inline docs/topics/compose-compiler-options.md:209]:
```suggestion
This lets Compose runtime suspend composition at skipping points,
splitting long-running compositions across multiple frames.
```

File: docs/topics/compose-compiler-options.md [hunk 4]
```diff
@@ -194,6 +194,27 @@ and functions that are implicitly not skippable (inline functions and functions
...
+
+Pausable composition is used in Lazy lists and other performance intensive components for prefetching content
```
Comment (sarahhaggarty) [inline docs/topics/compose-compiler-options.md:211]:
```suggestion
Lazy lists and other performance intensive components use pausable composition to prefetch content
```

File: docs/topics/compose-compiler-options.md [hunk 5]
```diff
@@ -194,6 +194,27 @@ and functions that are implicitly not skippable (inline functions and functions
 >
 {style="warning"}
 
+### PausableComposition
+
+> The feature flag affects behavior only with a version of Compose runtime that supports pausable composition,
```
Comment (sarahhaggarty) [inline docs/topics/compose-compiler-options.md:199]: Why not combine the info and warning into one warning at the top? If the warning is at the bottom of the text then readers can miss it if they don't read through all the content. Or think the warning is applicable to the next section. WDYT? :)

**Review by sarahhaggarty — COMMENTED** [review]
I made some suggestions to use active voice a bit more. Please let me know what you think about combining the notes :)

**Comment by zamulla** [comment]
These changes were cherry-picked into the 2.1.0 release branch, so closing the PR here.

**Pull request closed without merging**